cql2pgjson/src/main/java/org/z3950/zing/cql/cql2pgjson/CQL2PgJSON.java

```diff
@@ -594,6 +484,12 @@ private SqlSelect toSql(CQLSortNode node) throws QueryValidationException {
         break;
       }
 
+      // if number sort is specified explicitly
+      if (modifiers.cqlSortType == CqlSortType.NUMBER) {
+        order.append(vals.indexJson).append(desc);
+        continue;
+      }
+
```

Comment:
The switch clause and the if clause should be joined:

    if ("number".equals(vals.type) ||
        "integer".equals(vals.type) ||
        modifiers.cqlSortType == CqlSortType.NUMBER) {